Tasks/CopyFilesV2/copyfiles.ts

```diff
@@ -36,124 +37,196 @@ function makeDirP(targetFolder: string, ignoreErrors: boolean): void {
+        retryCount = 0;
+    }
+
+    if (isNaN(delayBetweenRetries) || delayBetweenRetries < 0) {
+        delayBetweenRetries = 0;
+    }
+
+    const preserveTimestamp: boolean = tl.getBoolInput('preserveTimestamp', false);
+    const ignoreMakeDirErrors: boolean = tl.getBoolInput('ignoreMakeDirErrors', false);
+
+    // normalize the source folder path. this is important for later in order to accurately
+    // determine the relative path of each found file (substring using sourceFolder.length).
+    sourceFolder = path.normalize(sourceFolder);
+    let allPaths: string[] = tl.find(sourceFolder, findOptions);
+    let sourceFolderPattern = sourceFolder.replace('[', '[[]'); // directories can have [] in them, and they have special meanings as a pattern, so escape them
+    let matchedPaths: string[] = tl.match(allPaths, contents, sourceFolderPattern); // default match options
+    let matchedFiles: string[] = matchedPaths.filter((itemPath: string) => !tl.stats(itemPath).isDirectory()); // filter-out directories
+
+    // copy the files to the target folder
+    console.log(tl.loc('FoundNFiles', matchedFiles.length));
+
+    if (matchedFiles.length > 0) {
+        // clean target folder if required
+        if (cleanTargetFolder) {
+            console.log(tl.loc('CleaningTargetFolder', targetFolder));
+
+            // stat the targetFolder path
+            let targetFolderStats: tl.FsStats;
+            try {
+                targetFolderStats = await retryHelper.RunWithRetrySingleArg<tl.FsStats, string>(
+                    () => tl.stats(targetFolder),
+                    targetFolder);
+            }
+            catch (err) {
+                if (err.code != 'ENOENT') {
+                    throw err;
+                }
             }
-        }
 
-        if (targetFolderStats) {
-            if (targetFolderStats.isDirectory()) {
-                // delete the child items
-                fs.readdirSync(targetFolder)
-                    .forEach((item: string) => {
+            if (targetFolderStats) {
+                if (targetFolderStats.isDirectory()) {
+                    // delete the child items
+                    const folderItems: string[] = await retryHelper.RunWithRetrySingleArg<string[], string>(
+                        () => fs.readdirSync(targetFolder),
+                        targetFolder);
+                    
+                    for (let item of folderItems) {
                         let itemPath = path.join(targetFolder, item);
-                        tl.rmRF(itemPath);
-                    });
-            }
-            else {
-                // targetFolder is not a directory. delete it.
-                tl.rmRF(targetFolder);
+                        await retryHelper.RunWithRetrySingleArg<void, string>(() => 
+                            tl.rmRF(itemPath),
+                            targetFolder
+                        );
+                    }
+                }
+                else {
```

Comment:
```suggestion
                } else {
````

Could you please check code formatting?

Reply:
fixed